pulp_container/tests/functional/utils.py

```diff
@@ -157,18 +157,16 @@ def gen_container_remote(url=REGISTRY_V2_FEED_URL, **kwargs):
     return gen_remote(url, upstream_name=kwargs.pop("upstream_name", REPO_UPSTREAM_NAME), **kwargs)
 
 
-def get_docker_hub_remote_blobsums(upstream_name=REPO_UPSTREAM_NAME):
+def get_blobsums_from_remote_registry(upstream_name=REPO_UPSTREAM_NAME):
     """Get remote blobsum list from dockerhub registry."""
-    token_url = (
-        "https://auth.docker.io/token"
-        "?service=registry.docker.io"
-        "&scope=repository:library/{0}:pull"
-    ).format(upstream_name)
+    token_url = "https://ghcr.io/token?service=ghcr.io&scope=repository:{0}:pull".format(
+        upstream_name
+    )
     token_response = requests.get(token_url)
     token_response.raise_for_status()
     token = token_response.json()["token"]
 
-    blob_url = ("{0}/v2/library/{1}/manifests/latest").format(REGISTRY_V2_FEED_URL, upstream_name)
+    blob_url = ("{0}/v2/{1}/manifests/latest").format(REGISTRY_V2_FEED_URL, upstream_name)
     response = requests.get(blob_url, headers={"Authorization": "Bearer " + token})
     response.raise_for_status()
     return response.json()["fsLayers"]
```

Comment:
this should be changed then respectively to 'layers'

Reply:
The GH Packages Registry does not support schema conversion. Thus, we cannot rely on `fsLayers` that was previously returned by DockerHub when we sent a request with no `accept` headers (which in result end up with converting manifest list from manifest.v2.list to manifest.v1).

Reply:
correct, that's why i am suggesting to change that to 'layers' because this is format for manifest.v2 version.
or you can leave fsLayers if you plan to push manifest.v1 schema.

Reply:
which when i look at your code you did exactly that :) https://github.com/pulp/pulp_container/pull/563/files#diff-092a0ab013f705709186cc205228a7603fe06485b818333a4225577bc843a0d6R181